File: 1636-number-of-substrings-with-only-1s/number-of-substrings-with-only-1s.c
#define MOD 1000000007
int numSub(char* s) {
    long int count=0;
    long int cur=0;
    int len=strlen(s);
    for(int i=0;i<len;i++){
        if(s[i]=='1'){
            cur++;
            count=(count+cur)%MOD;
        }else{
            cur=0;
        }
    }
    return count;
}

File: 1636-number-of-substrings-with-only-1s/number-of-substrings-with-only-1s.java
class Solution {
    public int numSub(String s) {
           long count = 0;
        long cur = 0;
        long MOD = 1000000007;
        int len = s.length();

        for(int i = 0; i < len; i++){
            if(s.charAt(i) == '1'){
                cur++;
                count = (count + cur) % MOD;
            } else {
                cur = 0;
            }
        }

        return (int) count;
    }
}